### PR TITLE
Revert "Bump semver from 6.3.0 to 7.0.0"

### PR DESCRIPTION
**Pull request by rwjblue**
This reverts commit 1c62ef668b3db753c5ff480365da779ead250c74.

**_tldr;_** `semver@7.x` _arguably_ includes breaking changes in a minor release (from 7.0.0 to 7.1.0), and we cannot update to 7.x until ember-cli-htmlbars drops Node 8 support.

At the time the `semver@7.0.0` version bump (from 1c62ef66) landed the CHANGELOG.md for `semver@7.0.0` stated:

* Refactor module into separate files for better tree-shaking
* Drop support for very old node versions, use const/let, => functions, and classes.

The verbiage here is obviously imprecise, and when merging (and releasing) I misinterpretted "very old node versions" as being versions older than Node 4.0.0 (in other words, versions that didn't support the features stated immediately following). Unfortunately, semver changed their `engines` version _after_ 7.0.0 was released (included in the 7.1.0 release) to only allow Node 10 or higher (see [here](https://github.com/npm/node-semver/commit/d61f828e64260a0a097f26210f5500e91a621828)).  Unfortunately, due to the way `yarn` works by default, this change in `engines` is actually a breaking change for `yarn` users (unless they specify the `--ignore-engines` flag).

### CODE REVIEW

**Comment by rwjblue** [comment]
Released in 4.2.2 https://github.com/ember-cli/ember-cli-htmlbars/releases/tag/v4.2.2
